src/apm-cli.js

```diff
@@ -0,0 +1,244 @@
+    help += "\n  colored output.";
+  }
+  console.error(help);
+};
+
+function printVersions(args, callback) {
+  const apmVersion = require('../package.json').version || '';
+  const npmVersion = require('npm/package.json').version || '';
+  const nodeVersion = process.versions.node || '';
+  getPythonVersion( pythonVersion =>
+    git.getGitVersion( gitVersion =>
+      getAtomVersion( atomVersion => {
+        if (args.json) {
+          const versions = {
+            apm: apmVersion,
+            npm: npmVersion,
+            node: nodeVersion,
+            atom: atomVersion,
+            python: pythonVersion,
+            git: gitVersion,
+            nodeArch: process.arch
+          };
+          if (config.isWin32()) {
+            versions.visualStudio = config.getInstalledVisualStudioFlag();
+          }
+          console.log(JSON.stringify(versions));
+        } else {
+          const versions = [
+            `apm  ${apmVersion}`.red,
+            `npm  ${npmVersion}`.green,
+            `node ${nodeVersion} ${process.arch}`.blue,
+            `atom ${atomVersion || ''}`.cyan,
+            `python ${pythonVersion || ''}`.yellow,
+            `git ${gitVersion || ''}`.magenta
+          ];
+          if (config.isWin32()) {
+            const visualStudioVersion = config.getInstalledVisualStudioFlag() || '';
+            versions.push(`visual studio ${visualStudioVersion}`.cyan);
+          }
+          console.log(versions.join("\n"));
+        }
+        callback();
+      })
+    )
+  );
+};
+
+function getAtomVersion(callback) {
+  config.getResourcePath( resourcePath => {
+    const unknownVersion = 'unknown';
+    try {
+      const version = require(path.join(resourcePath, 'package.json')).version || unknownVersion;
+      callback(version);
+    } catch (e) {
+      callback(unknownVersion);
+    }
+  });
+};
+
+function getPythonVersion(callback) {
+  const npmOptions = {
+    userconfig: config.getUserConfigPath(),
+    globalconfig: config.getGlobalConfigPath()
+  };
+  npm.load(npmOptions, () => {
+    let python = npm.config.get('python') || process.env.PYTHON;
+    if (config.isWin32() && !python) {
+      let rootDir = process.env.SystemDrive || 'C:\\';
+      if (rootDir[rootDir.length - 1] !== '\\') {
+        rootDir += '\\';
+      }
+      const pythonExe = path.resolve(rootDir, 'Python27', 'python.exe');
+      if (fs.isFileSync(pythonExe)) {
+        python = pythonExe;
+      }
+    }
+    if (!python) python = 'python';
+    const spawned = spawn(python, ['--version']);
+    const outputChunks = [];
+    spawned.stderr.on('data', chunk =>
+      outputChunks.push(chunk)
+    );
+    spawned.stdout.on('data', chunk =>
+      outputChunks.push(chunk)
+    );
+    spawned.on('error', () => {});
+    spawned.on('close', code => {
+      let version;
+      if (code === 0) {
+        version = Buffer.concat(outputChunks).toString().split(' ')[1]?.trim();
```

Comment:
Love the use of an optional chaining operator to `trim()` the `version` here.

My only concern is the original decaffienated code produces 

```javascript
[name, version] =Array.from(Buffer.concat(outputChunks).toString().split(' '));
version = version != null ? version.trim() : undefined;
```

So while the `trim()` is added perfectly the defaulting behavior to `undefined` isn't retained. I'm not sure if this is used later on or not, but maybe should be addressed? We could add a Nullish coalescing operator to the end to check for `null` or `undefined`

```javascript
version = Buffer.concat(outputChunks).toString().split(' ')[1]?.trim() ?? undefined;
```